Backends/CLX/clipboard.lisp

```diff
@@ -0,0 +1,285 @@
+                                                  object-type)
+                (declare (ignore port object-type))
+                (let* ((window (sheet-direct-xmirror (climb:selection-object-owner object)))
+                       (display (xlib:window-display window)))
+                  ;; We're not actually supposed to call set-selection-owner without
+                  ;; a timestamp due to the following statemnt in ICCCM:
+                  ;;
+                  ;;     Clients attempting to acquire a selection must set the time
+                  ;;     value of the SetSelectionOwner request to the timestamp of
+                  ;;     the event triggering the acquisition attempt, not to
+                  ;;     CurrentTime. A zero-length append to a property is a way to
+                  ;;     obtain a timestamp for this purpose; the timestamp is in
+                  ;;     the corresponding PropertyNotify event.
+                  ;;
+                  ;; The reasons for his seems to be to ensure that the ownership is
+                  ;; actually transferred correctly. This shouldn't be a major issue
+                  ;; in practice, and it significantly simplifies the implementation.
+                  (xlib:set-selection-owner display selection window nil)
+                  (if (eq (xlib:selection-owner display selection) window)
+                      (call-next-method)
+                      (warn "Couldn't set X11 selection ~s owner to ~s."
+                            selection window))))))
+  (frob :primary)
+  (frob :clipboard))
+
+(macrolet ((frob (selection)
+             `(defmethod climb:release-selection ((port clx-selection-mixin)
+                                                  (selection (eql ,selection))
+                                                  (object climb:selection-object))
+                (alexandria:when-let*
+                    ((sheet (climb:selection-object-owner object))
+                     (window (sheet-direct-xmirror sheet)))
+                  (xlib:set-selection-owner (xlib:window-display window) selection nil nil))
+                (call-next-method))))
+  (frob :primary)
+  (frob :clipboard))
+
+;;;  Paste support
+;;;
+;;;  When a client requests the content of the clipboard or selection, a REQUESTED-OBJECT
+;;;  parameter is used to indicate what kind of of object the caller expects to
+;;;  receive. Selection objects are explained in ports.lisp.
+;;;
+;;;  First we shortcut result if the selection owner is a sheet in the same image. Then we
+;;;  try to negotiate the type – ask for :TARGETS. If response is empty we default to
+;;;  :STRING. Then we verify whether we can translate proposed types to the requested one
+;;;  and if it is possible we request that type with convert-selection. Otherwise NIL is
+;;;  returned.
+
+(defun clx-request-selection (port selection requested-object)
+  ;; Shortcut the request if the selection is owned by a window in the same
```

Comment:
Shouldn't this comment be in the `REQUEST-SELECTION` method instead? That's where the actual shortcut happens.

Reply:
you are right, omission (fixed)